BedrockCore.cpp

```diff
@@ -28,6 +28,36 @@ class AutoScopeRewrite {
     bool (*_handler)(int, const char*, string&);
 };
 
+uint64_t BedrockCore::_getTimeout(const SData& request) {
+    uint64_t timeout = request.isSet("timeout") ? request.calc("timeout") : DEFAULT_TIMEOUT;
+
+    if (timeout > 2'000'000) {
+        // Old microsecond timeout. Update caller to use milliseconds. Remove this line once we no longer see this.
+        SWARN("[TYLER] old-style timeout found for command: " << request.methodLine);
+        timeout /= 1000;
+    }
+
+    // See when the command was scheduled to run. The timeout is from *this* start time, not from when the command
```

Comment:
I'm a bit confused.  Let's say this command is run on a slave and then escalated to master.  Is `commandExecuteTime` when it started on the slave and `timeout` is now?

Reply:
`commandExecuteTime` is when it was supposed to start on the slave. `timeout` is a duration (default is 30 seconds).

So if a command was scheduled to run on a slave at `12:30:10` and had a default timeout, then when the command is escalated to master, it still needs to finish by `12:30:40` or it will time out.